ui/details-panel/SimilarJobsTab.jsx

```diff
@@ -0,0 +1,302 @@
+    super(props);
+
+    const { $injector } = this.props;
+    this.ThJobModel = $injector.get('ThJobModel');
+    this.$rootScope = $injector.get('$rootScope');
+    this.ThTextLogStepModel = $injector.get('ThTextLogStepModel');
+    this.ThResultSetModel = $injector.get('ThResultSetModel');
+    this.thNotify = $injector.get('thNotify');
+    this.thTabs = $injector.get('thTabs');
+    this.thClassificationTypes = $injector.get('thClassificationTypes');
+
+    this.pageSize = 20;
+    this.tab = this.thTabs.tabs.similarJobs;
+
+    this.state = {
+      similarJobs: [],
+      filterBuildPlatformId: true,
+      filterOptionCollectionHash: true,
+      page: 1,
+      selectedSimilarJob: null,
+      hasNextPage: false,
+      selectedJob: null,
+      isLoading: true,
+    };
+
+    // map between state fields and job fields
+    this.filterMap = {
+      filterBuildPlatformId: 'build_platform_id',
+      filterOptionCollectionHash: 'option_collection_hash',
+    };
+  }
+
+  componentDidMount() {
+    this.getSimilarJobs = this.getSimilarJobs.bind(this);
+    this.showNext = this.showNext.bind(this);
+    this.toggleFilter = this.toggleFilter.bind(this);
+
+    this.jobClickUnlisten = this.$rootScope.$on(thEvents.jobClick, (event, job) => {
+      this.setState({ selectedJob: job, similarJobs: [], isLoading: true }, this.getSimilarJobs);
+    });
+  }
+
+  componentWillUnmount() {
+    this.jobClickUnlisten();
+  }
+
+  async getSimilarJobs() {
+    const { page, selectedJob, similarJobs, selectedSimilarJob } = this.state;
+    const { repoName } = this.props;
+    const options = {
+      // get one extra to detect if there are more jobs that can be loaded (hasNextPage)
+      count: this.pageSize + 1,
+      offset: (page - 1) * this.pageSize
+    };
+
+    ['filterBuildPlatformId', 'filterOptionCollectionHash']
+      .forEach((key) => {
+        if (this.state[key]) {
+          const field = this.filterMap[key];
+          options[field] = selectedJob[field];
+        }
+    });
+
+    const newSimilarJobs = await this.ThJobModel.get_similar_jobs(repoName, selectedJob.id, options);
+
+    if (newSimilarJobs.length > 0) {
+      this.setState({ hasNextPage: newSimilarJobs.length > this.pageSize });
+      newSimilarJobs.pop();
+      // create an array of unique push ids
+      const pushIds = [...new Set(newSimilarJobs.map(job => job.result_set_id))];
+      // get pushes and revisions for the given ids
+      const pushListResp = await this.ThResultSetModel.getResultSetList(repoName, pushIds, true);
+      const pushList = pushListResp.data;
+      //decorate the list of jobs with their result sets
+      const pushes = pushList.results.reduce((acc, push) => (
+        { ...acc, [push.id]: push }
+      ), {});
+      newSimilarJobs.forEach((simJob) => {
+        simJob.result_set = pushes[simJob.result_set_id];
+        simJob.revisionResultsetFilterUrl = `/#/jobs?repo=${repoName}&revision=${simJob.result_set.revisions[0].revision}`;
```

Comment:
Perhaps these could be generated via a modified `jobsUrl()` in `urlHelper.js`?

Reply:
Another good call.   Done